ukvm/ukvm_elf.c

```diff
@@ -197,6 +217,17 @@ void ukvm_elf_load(const char *file, uint8_t *mem, size_t mem_size,
             if (mprotect(daddr, _end - paddr, PROT_EXEC | PROT_READ) == -1)
                 goto out_error;
         }
+
+#ifdef __aarch64__
+        ukvm_region_list->regions[ph_i].slot = ph_i;
+        ukvm_region_list->regions[ph_i].guest_phys_addr = (uint64_t)paddr;
+        ukvm_region_list->regions[ph_i].memory_size = _end - paddr;
+        ukvm_region_list->regions[ph_i].userspace_addr = (uint64_t)daddr;
+        ukvm_region_list->regions[ph_i].flags = (phdr[ph_i].p_flags & PF_X)
+                                                ? KVM_MEM_READONLY
+                                                : 0;
+        ukvm_region_list->count++;
+#endif
```

Comment:
Hi Mato,
Your understanding about ukvm_region_list is right.
But the section address and size like .text is not fixed. It will change as we add or remove code.
About the mprotect, we had tried it already. But mapping a read-only VMA is only allowed if the
memory region is configured as read-only. If we set all guest memory to RWX, the mprotect will
failed when we try to make an area be read-only.